### PR TITLE
[skipruntime] Manage external resources reuse

### PR DESCRIPTION
**Pull request by skiplabsdaniel**

### CODE REVIEW

File: skiplang/prelude/src/stdlib/debug/Debug.sk [hunk 1]
```diff
@@ -55,7 +55,7 @@ private fun debugImpl<T>(x: T, print: String -> void): void {
  *
  * For your class and return one of the values here, as appropriate.
  */
-base class .Inspect uses Hashable {
+base class .Inspect uses Hashable, Equality {
```
Comment (beauby) [inline skiplang/prelude/src/stdlib/debug/Debug.sk]: Why is this needed?

Comment (skiplabsdaniel) [reply skiplang/prelude/src/stdlib/debug/Debug.sk]: I'm investigating on that.

Comment (beauby) [reply skiplang/prelude/src/stdlib/debug/Debug.sk]: Did you figure out the problem? If not, remember to make an issue summing up your investigation.

Comment (skiplabsdaniel) [reply skiplang/prelude/src/stdlib/debug/Debug.sk]: No, I don't. It's on my plan.
